modules/improvedigitalBidAdapter.js

```
@@ -276,41 +272,86 @@ const ID_REQUEST = {
+      if (extendMode) {
+        return EXTEND_URL;
+      }
+
+      const urlSegments = [];
+      urlSegments.push(hasPurpose1Consent(bidderRequest?.gdprConsent) ? AD_SERVER_BASE_URL : BASIC_ADS_BASE_URL)
+      if (publisherId) {
+        urlSegments.push(publisherId)
+      }
+      urlSegments.push(PB_ENDPOINT)
+
+      return urlSegments.join('/');
+    }
+
+    function formatRequest(imps, {transactionId, publisherId}, extendMode) {
       const request = deepClone(basicRequest);
       request.imp = imps;
       request.id = getUniqueIdentifierStr();
+      setAdditionalConsent(request, extendMode);
       if (transactionId) {
         deepSetValue(request, 'source.tid', transactionId);
       }
-      const adServerUrl = hasPurpose1Consent(bidderRequest?.gdprConsent) ? AD_SERVER_URL : BASIC_ADS_URL;
+
       return {
         method: 'POST',
-        url: extendMode ? EXTEND_URL : adServerUrl,
+        url: adServerUrl(extendMode, publisherId),
         data: JSON.stringify(request),
         bidderRequest
       }
     };
 
+    let publisherId = null;
     bidRequests.map((bidRequest) => {
-      const extendModeEnabled = this.isExtendModeEnabled(globalExtendMode, bidRequest.params);
+      const bidParam = bidRequest.params;
+      const extendModeEnabled = this.isExtendModeEnabled(globalExtendMode, bidParam);
       const imp = this.buildImp(bidRequest, extendModeEnabled);
+      publisherId = bidParam?.publisherId;
       if (singleRequestMode) {
         extendModeEnabled ? extendImps.push(imp) : adServerImps.push(imp);
       } else {
-        requests.push(formatRequest([imp], bidRequest.transactionId, extendModeEnabled));
+        requests.push(formatRequest([imp], {transactionId: bidRequest.transactionId, publisherId}, extendModeEnabled));
       }
     });
 
     if (!singleRequestMode) {
       return requests;
     }
+    const requestOptions = {
+      transactionId: bidderRequest.auctionId,
+      publisherId,
```

In singleRequestMode all placements in the request must have the same publisherId. Otherwise you should throw an error.

---

Okay